packages/core/CalendarManager.ts

```diff
@@ -48,16 +56,29 @@ export const getConnectedCalendars = async (
         }
         const cals = await calendar.listCalendars();
         const calendars = _(cals)
-          .map((cal) => ({
-            ...cal,
-            readOnly: cal.readOnly || false,
-            primary: cal.primary || null,
-            isSelected: selectedCalendars.some((selected) => selected.externalId === cal.externalId),
-            credentialId,
-          }))
+          .map((cal) => {
```

Comment:
Since we are looping through all calendars, let's find both the primary and destination calendar